src/ComicK/ComicKParser.ts

```diff
@@ -118,86 +118,96 @@ export const parseChapters = (
     strictNameMatching: boolean,
     uploaders: string[]
 ): void => {
+    var filteredChapters = data.chapters
     if (uploadersAutoFiltering) {
-        const _chapters = new Map<number, { score: number, chapter: ChapterData }>()
-        for (const chapter of data.chapters) {
+        filteredChapters = filterUploadersByScore(data.chapters)
```

Comment:
Also types will be wrong since data.chapter is of different type compared to return type of function here

Reply:
yep data.chapter is `ChapterData`, which is what I'm using for filtering since that's what has the score, and then on line 127 we convert those to `Chapter`s and push them to the `chapters` array that is being passed in. I'm pretty sure the types are correct since it compiles fun and seems to work fine on my phone. I tested:
- Autofiltering
- Whitelisting with strict name matching
- Whitelisting without strict name matching
- Blacklisting with strict name matching
- Blacklisting without strict name matching

(though I did change some logic around strict name matching, see the other comment I just made)

Reply:
I suggest the following:

```ts
    const chaptersData: ChapterData[] = []
    if (uploadersAutoFiltering) {
        filterUploadersByScore(data.chapters, chapterData) // returns void
    } else {
        filteredChapters = filterUploadersByList(data.chapters, uploadersToggled, uploadersWhitelisted, aggressiveUploadersFilter, strictNameMatching, uploaders, chapterData) // returns void
    }
```

Reply:
Also consider using a better description for `uploadersAutoFiltering`, perhaps `uploadersScoreFiltering`, or even `chapterScoreFiltering` because it relates more to individual chapters more than uploaders.

Reply:
sure, I can do make those changes

Reply:
updated